lib/spack/spack/cmd/testsuite.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+
+import os
+import requests
+import glob
+import argparse
+import llnl.util.tty as tty
+from spack.spec import Spec
+import spack.cmd.install as install    
+import spack.cmd.uninstall as uninstall
+import spack.util.spack_yaml as yaml
+import spack.compilers 
+import spack
+
+description = "Compiles a list of tests from a yaml file. Runs Spec and concretize then produces cdash format."
+
+#not sure if I will use this later cdash or junit, spack.io or whereever or other flags
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-c', '--complete', action='store_true', dest='complete',
+        help='using this option switches from simple cdash output to compelet: simple is only build, complete is configure, build and test xml output.')
+    subparser.add_argument(
+        'yamlFile', nargs=argparse.REMAINDER,
+        help="yaml file that contains a list of tests, example yaml file can be found in /lib/spack/docs/tutorial/examples/test.yaml")
+
+'''
+fucntion Name: reduceCompilerList
+parameters:
+    masterList : list
+returns:
+    list
+Get the list of compilers from spack found on the system.
+Compares the spack list to test file.
+If a compiler is found on both lists its returned.
+'''
+def reduceCompilerList(masterList):
+    systemCompilers = []
+    newMaster = []
+    newList = []
+    for item in masterList:
+        newMaster.append(str(item))
+    
+    for compilers in spack.compilers.all_compilers(scope=None):
+        systemCompilers.append(str(compilers))
+
+    for compiler in systemCompilers:
+        if compiler in newMaster:
+            newList.append(compiler)
+    return newList
+
+
+'''
+fucntion Name: removeTests
+parameters:
+    testList  : list
+    exclusion : str
+returns:
+    list
+
+Receieves a list of all tests and an excusion.
+Depending on the type of exclusion it will remove tests matching.
+Returns the list with the tests matching the excusion removed.
+
+exclusions can be in this form:
+pkg%compiler
+pkg@version
+compiler@version
+pkg@version%compiler@version
+pkg@version%compiler
+pkg%compiler@version
+pkg
+compiler
+'''
+def removeTests(testList, exclusion):
+    pkg = ""
+    compiler=""
+    teststoRemove = []
+    if "%" in exclusion:
```

Comment:
Im not sure this is what I am doing. This is for exclusion, by doing the string analysis so I can determine how to properly test for the exclusion in the set of tests.